framework/src/mesh/MooseMesh.C

```diff
@@ -1281,6 +1288,10 @@ MooseMesh::detectPairedSidesets()
   {
     Elem * elem = *el;
 
+    // dimension of the current element and its normals
+    unsigned int i = elem->dim() - 1;
```

Comment:
Maybe call this variable something more useful... like `side_dim`?  I actually had trouble figuring out what was going on down below...